ipld-gateways.md

```diff
@@ -0,0 +1,145 @@
+Similarly the gateway will informus what the encoding is for the data we asked for.
+-->
+
+- `GET /ipld/cid/supath`
+- `Accept: application/json`
+- `Accept: application/vnd.ipld.dag-cbor`
+- Default to CID's encoding?
+- Respond with `Content-Type: application/json`, or encoding used
+- Similar to `?format` parameter in `GET /ipfs/`
+
+## Path Segment Parameters
+
+<!--
+A slightly more advanced feature, which makes raw IPLD different from IPFS paths, is the ability to view data through different lenses.
+I propose we take full advantage of this in gateways and make it easy for applications to signal that they would like to use ADLs and the such.
+With this we can apply ADLs, schemas, and selectors for segments in the IPLD path.
+-->
+
+- IPLD has "lenses" for viewing data
+- Can we add them to segments of a path?
+- `/ipld/cid/[ADL=HAMT]example`
+- Alternately, specify info in the querystring for the root CID?
+
+### Syntax
+
+<!--
+Each `segment` in an IPLD path can have some parameters surrounded by `[` and `]`.
+The content of these parameters follows the same encoding as query strings, where parameters specify lenses and IPLD transofmrations that should be made on the data before returning it to the client.
+-->
+
+- Segments of a path can have parameters
+- `[` and `]` for surrounding parameters
+- Parameters use same syntax as querystrings
+- e.g. `[ADL=foo&Schema=bar]`
+- Extracted out of the path specifier
+- Use URL encoding for escaping new special characters.
+
+### ADLS/Schemas/Selectors?
+
+<!--
+So, I'd like to reserve this parameter syntax sooner than later, but I'd like some feedback before suggesting specific ways that parameters should be handled.
+I've put some questions together that it'd be useful to contemplate and to record answers from folks in the room.
+Feel free to bring up related points here too.
+-->
+
+- What ADLs should we support? (HAMT to start?)
+- How should schemas be linked to? (URL to IPLD data for schema? Inline JSON?)
+- Is there a good way to serialize selectors?
+
+## Writing
+
+<!--
+So, reading IPLD data is a good start, but I think that authoring content is just as important if not more so important for p2p tech to really spread.
+With that in mind, I wanted to apply some of the learnings from writable gateways for IPFS and IPNS to IPLD.
+-->
+
+- Writing data is important! Not just reading.
+- Learned from writable gateway requirements
+
+### Basic
+
+<!--
+Uploading IPLD data could be similar to IPFS.
+We can POST data encoded in whatever format we're used to to `/ipld/` and get back the CID for it.
+Or we can PUT some data over top of an existing CID with some data and get back an updated root.
+As well, similar to being able to read data in different formats, we can hint to the gateway which encoding we're using for IPLD.
+One thing that remains is to figure out how we could POST data in one encoding, but have the gateway save it in another one.
```

Comment:
Added line about how this can be useful for when JSON is easier to use as a library, but would want the actual stored data to be more efficient like dag-cbor